source/Calamari.Terraform/TerraformCliExecutor.cs

```diff
@@ -0,0 +1,212 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Calamari.Deployment;
+using Calamari.Extensions;
+using Calamari.Integration.FileSystem;
+using Calamari.Integration.Processes;
+using Calamari.Integration.Proxies;
+using Octopus.CoreUtilities.Extensions;
+
+namespace Calamari.Terraform
+{
+    class TerraformCLIExecutor : IDisposable
+    {
+        private readonly ICalamariFileSystem fileSystem;
+        private readonly RunningDeployment deployment;
+        readonly Dictionary<string, string> environmentVariables;
+        private Dictionary<string, string> defaultEnvironmentVariables;
+        private readonly string logPath;
+
+        public TerraformCLIExecutor(ICalamariFileSystem fileSystem, RunningDeployment deployment,
+            Dictionary<string, string> environmentVariables)
+        {
+            this.fileSystem = fileSystem;
+            this.deployment = deployment;
+            this.environmentVariables = environmentVariables;
+
+            var variables = deployment.Variables;
+            TemplateDirectory = variables.Get(TerraformSpecialVariables.Action.Terraform.TemplateDirectory, deployment.CurrentDirectory);
+            Workspace = variables.Get(TerraformSpecialVariables.Action.Terraform.Workspace);
+            InitParams = variables.Get(TerraformSpecialVariables.Action.Terraform.AdditionalInitParams);
+            ActionParams = variables.Get(TerraformSpecialVariables.Action.Terraform.AdditionalActionParams);
+            TerraformExecutable = variables.Get(TerraformSpecialVariables.Action.Terraform.CustomTerraformExecutable) ??
+                                  $"terraform{(CalamariEnvironment.IsRunningOnWindows ? ".exe" : String.Empty)}";
+            AllowPluginDownloads = variables.GetFlag(TerraformSpecialVariables.Action.Terraform.AllowPluginDownloads, true);
+            AttachLogFile = variables.GetFlag(TerraformSpecialVariables.Action.Terraform.AttachLogFile);
+            TerraformVariableFiles = GenerateVarFiles();
+
+            logPath = Path.Combine(deployment.CurrentDirectory, "terraform.log");
+
+            if (!String.IsNullOrEmpty(TemplateDirectory))
+            {
+                var templateDirectoryTemp = Path.Combine(deployment.CurrentDirectory, TemplateDirectory);
+
+                if (!Directory.Exists(templateDirectoryTemp))
+                {
+                    throw new Exception($"Directory {TemplateDirectory} does not exist.");
+                }
+
+                TemplateDirectory = templateDirectoryTemp;
+            }
+
+            InitialiseTerraformEnvironmentVariables();
+
+            InitialisePlugins();
```

Comment:
Initiali`z`ePlugins (same above\below)

Reply:
Do we prefer American spelling for internal apis?

Reply:
we tend to use it throughout most of the code. just a consistency thing

Reply:
Yeah, US English spelling for code generally. It's a standing code convention.

Reply:
fixed